### DISCUSSION TITLE
Capture and backreference

### DISCUSSION THREAD

**Discussion by jimmydevine** [taocpp/PEGTL#336] [Q&A] [answered]
Is it possible to build a grammar with xml tag name matching, ie <**tag**>...</**tag**> but not match <**tag**>...</**other**>
Something similar to the following regex /<(**[^>]+**)>.*?<\\/**\1**>/

**Comment by skyrich62** [comment, marked as answer]
Yes.  In fact there's a very good explanation of the technique required and so-called "raw string" matching in the docs, and in the examples.

.../contrib/raw_string.hpp   is a good place to start.   See the docs at .../doc/Contrib-and-Examples.  The example there is a lua raw string, but once you understand what's happening there, it can be modified for xml.

**Reply by ColinH** [reply]
A simplified version of raw_string can be found in the [dynamic_match](https://github.com/taocpp/PEGTL/blob/main/src/example/pegtl/dynamic_match.cpp) example.

**Comment by jimmydevine** [comment]
I was able to use this along with a stack to track proper end tags, thanks!
